src/bun.js/bindings/sqlite/JSSQLStatement.cpp

```diff
@@ -1526,14 +1585,35 @@ JSC_DEFINE_HOST_FUNCTION(jsSQLStatementPrepareStatementFunction, (JSC::JSGlobalO
     int64_t currentMemoryUsage = sqlite_malloc_amount;
 
     int rc = SQLITE_OK;
+    const char* tail = nullptr;
+    bool hasMultipleStatements = false;
+
     if (
         // fast path: ascii latin1 string is utf8
         sqlString.is8Bit() && simdutf::validate_ascii(reinterpret_cast<const char*>(sqlString.span8().data()), sqlString.length())) {
-        rc = sqlite3_prepare_v3(db, reinterpret_cast<const char*>(sqlString.span8().data()), sqlString.length(), flags, &statement, nullptr);
+        const char* sqlData = reinterpret_cast<const char*>(sqlString.span8().data());
+        const char* sqlEnd = sqlData + sqlString.length();
+        rc = sqlite3_prepare_v3(db, sqlData, sqlString.length(), flags, &statement, &tail);
+
+        // Check if there are remaining statements
+        if (rc == SQLITE_OK && tail != nullptr && tail < sqlEnd) {
```

Comment:
By design, multiple statements in sqlite3_prepare_v3 is not supported. Should this code still be here?